source/agora/node/Ledger.d

```diff
@@ -13,11 +13,46 @@
 
 module agora.node.Ledger;
 
+import agora.common.API;
 import agora.common.Block;
 import agora.common.Data;
 import agora.common.Hash;
 import agora.common.Transaction;
 
+import std.algorithm;
+import std.range;
+
+/// Input range of Blocks
+public struct BlockRange
+{
+    ///
+    private Block[Hash] ledger;
```

Comment:
I think we should move storage to be `Block*[Hash]` and backed by a contiguous list of items, possibly a memory mapped file. Because accessing by height seems to be a very, very common use case, while hash is less common.

Reply:
Did you mean `Block*[ulong]` then if you mean accessing by height?

Reply:
Well I suppose we should have both Hash and height lookups..

Reply:
Yeah, accessing by height with an AA lookup could be useful as well. The main point is that it needs to be backed by a contiguous storage, otherwise that's never going to scale. Unless you have other suggestions ?

Reply:
I'll think up of something.